source/grid/manifold.cc

```diff
@@ -926,14 +926,23 @@ FlatManifold<dim, spacedim>::normal_vector(
       xi += delta_xi;
       ++iteration;
 
+      // It turns out that the check in reference coordinates with an absolute
+      // tolerance can cause a convergence failure of the Newton method as
+      // seen in tests/manifold/flat_manifold_09.cc. To work around this, also
+      // use a convergence check in world coordinates. This check has to be
+      // relative to the size of the face of course. Here we decided to use
+      // diameter because it works for non-planar faces and is cheap to
+      // compute:
+      const double delta = (F - p).norm() / face->diameter();
```

Comment:
I got completely confused why you are able to use the same tolerance for the reference and the world coordinates. What I had missed is that you are normalizing here. Maybe call this `normalized_delta` or `relative_delta`?